browser/src/control/Control.AutoFillPreviewTooltip.ts

```diff
@@ -0,0 +1,129 @@
+/* -*- js-indent-level: 8 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+/*
+ * Control.AutoFillPreviewTooltip - class for tooltip of cell previews during auto fill
+ */
+
+/* global app */
+
+class AutoFillPreviewTooltip extends L.Control.AutoCompletePopup {
+	usageText: string;
+	newPopupData: PopupData;
+
+	constructor(map: ReturnType<typeof L.map>) {
+		super('autoFillPreviewTooltip', map);
+		this.newPopupData = {
+			children: [
+				{
+					id: 'container',
+					type: 'container',
+					enabled: true,
+					children: new Array<WidgetJSON>(),
+					vertical: true,
+				} as any as WidgetJSON,
+			] as Array<WidgetJSON>,
+			jsontype: 'dialog',
+			type: 'modalpopup',
+			cancellable: false,
+			popupParent: '_POPOVER_',
+			clickToClose: '_POPOVER_',
+			id: 'autoFillPreviewTooltip',
+			title: 'HELLO',
+		} as PopupData;
+	}
+
+	onAdd() {
+		this.newPopupData.isAutoCompletePopup = true;
+		this.map.on('openautofillpopup', this.openAutoFillPopup, this);
+		this.map.on('closementionpopup', this.closeMentionPopup, this);
+		this.map.on('sendautofilllocation', this.sendAutoFillLocation, this);
+		// this.map.on('mousedown', this.onMouseDown, this);
+		// this.map.on('mouseup', this._onMouseUp, this);
+	}
+
+	// private onMouseUp (e: MouseEvent) {
+	//     console.error("yaz: " + e);
+	// }
+
+	getAutoFillAreaBottomRightLocation(ev: FireEvent): Point {
+		// var currPos = ev.data.location;
+		var currPos = {
+			// x: app.file.textCursor.rectangle.cX1,
+			x: app.file.viewedRectangle.cX1,
+			y: app.file.textCursor.rectangle.cY2,
```

Comment:
textCursor -> where "caret" is while editing text inside cell
cellCursor -> position of selected cell

please look into `docstate.js`

Reply:
I use:
```ts

var currPos = {
	x:
		app.calc.cellCursorRectangle.pWidth +
		app.calc.cellCursorRectangle.pX1 +
		5,
	y:
		app.calc.cellCursorRectangle.pHeight +
		app.calc.cellCursorRectangle.pY1 +
		5,
};

```

### currently we have: 
https://github.com/user-attachments/assets/35591294-fa46-4348-9200-74553ef59100

### issues
- Popup width is too long
- popup location should be updated per autofill selection

Reply:
well, at least we can move the popup right now. I needed to close the popup before update it :)

https://github.com/user-attachments/assets/747dfd19-5705-48e7-b27e-374be7a72f85

Reply:
Popup location is now fine. I'm going to fix the Popup width.